app/models/account.rb

```diff
@@ -36,18 +40,56 @@ class Account < ApplicationRecord
   # @return [Account] A persisted instance of Account
   #
   def self.cache_or_create_by!(user)
-    return unless user.uuid
+    return unless user.uuid || user.sec_id
 
-    do_cached_with(key: user.uuid) do
+    # if possible use the idme uuid for the key, fallback to using the sec id otherwise
+    key = user.uuid || "sec:#{user.sec_id}"
+    acct = do_cached_with(key: key) do
       create_if_needed!(user)
     end
+    # Account.sec_id was added months after this class was built, thus
+    # the existing Account records (not new ones) need to have their
+    # sec_id value updated
+    update_if_needed!(acct, user)
   end
 
   def self.create_if_needed!(user)
-    find_or_create_by!(idme_uuid: user.uuid) do |account|
-      account.edipi = user&.edipi
-      account.icn   = user&.icn
+    attrs = account_attrs_from_user(user)
+
+    accts = where(idme_uuid: attrs[:idme_uuid])
+            .where.not(idme_uuid: nil)
+            .or(
+              where(sec_id: attrs[:sec_id])
+              .where.not(sec_id: nil)
+            )
+
+    if accts.length > 1
+      data = accts.map(&:attributes)
+      log_message_to_sentry('multiple Account records with matching ids', 'warning', data)
     end
+
+    accts.length.positive? ? accts[0] : create(**attrs)
+  end
+
+  def self.update_if_needed!(account, user)
+    # account has yet to be saved, no need to update
+    return account unless account.id
```

Comment:
```suggestion
    return account unless account.persisted?
```
Does `persisted?` work here instead? Seems easier to read and could warrant removing the above comment.